src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -0,0 +1,167 @@
+  // where origin is the source of a connect. note that if the source is a trivial primop that
+  // has yet to be optimized via constant propagation, the function will try to search backwards
+  // past the primop. this allows you to compare if two nodes have the same origin, for example. 
+  // backward searches until PrimOp, Lit or non-trivial Mux appears
+  // technically, you should keep searching through PrimOp, because a node + 0 is still itself, 
+  // a node shifted by 0 is still itself, etc. 
+  // TODO: handle validif???, more thorough
+  private def getConnectOrigin(connects: Map[String, Expression], e: Expression): Expression = e match { 
+    case Mux(cond, tv, fv, _) =>
+      val fvOrigin = getOrigin(connects,fv)
+      val tvOrigin = getOrigin(connects,tv)
+      val condOrigin = getOrigin(connects,cond)
+      if (we(tvOrigin) == we(one) && we(fvOrigin) == we(zero)) condOrigin
+      else if (we(condOrigin) == we(one)) tvOrigin
+      else if (we(condOrigin) == we(zero)) fvOrigin
+      else if (we(tvOrigin) == we(fvOrigin)) tvOrigin
+      else if (we(fvOrigin) == we(zero) && we(condOrigin) == we(tvOrigin)) condOrigin
+      else e
+    case DoPrim(PrimOps.Or, args, consts, tpe) if args.contains(one) => one
+    case DoPrim(PrimOps.And, args, consts, tpe) if args.contains(zero) => zero
+    case DoPrim(PrimOps.Bits, args, consts, tpe) =>  
+      val msb = consts(0)
+      val lsb = consts(1) 
+      val extractionWidth = (msb-lsb)+1
+      val nodeWidth = bitWidth(args.head.tpe)
+      // if you're extracting the full bitwidth, then keep searching for origin
+      if (nodeWidth == extractionWidth) getOrigin(connects,args.head)
+      else e
+    case DoPrim(op, args, _, _) if (op == PrimOps.AsUInt || op == PrimOps.AsSInt || op == PrimOps.AsClock) =>
+      getOrigin(connects,args.head)
+    case _: WRef | _: SubField | _: SubIndex | _: SubAccess if connects contains e.serialize =>
+      getConnectOrigin(connects,e.serialize) 
+    case _ => e
+  }
+
+  def appendInfo[T <: Info](info: T, add: Map[String,Any]) = info match {
+    case i: AppendableInfo => i.append(add)
+    case _ => AppendableInfo(fields = add + ("info" -> info))
+  }
+  def appendInfo[T <: Info](info: T, add: (String,Any)): AppendableInfo = appendInfo(info,Map(add))
+  def getInfo[T <: Info](info: T, k: String) = info match{
+    case i: AppendableInfo => i.get(k)
+    case _ => None
+  }
+  def containsInfo[T <: Info](info: T, k: String) = info match {
+    case i: AppendableInfo => i.fields.contains(k)
+    case _ => false
+  }
+
+  def eqMems(a: DefMemory, b: DefMemory) = {
+    a.info == b.info &&
+    a.dataType == b.dataType &&
+    a.depth == b.depth &&
+    a.writeLatency == b.writeLatency &&
+    a.readLatency == b.readLatency &&
+    a.readers == b.readers &&
+    a.writers == b.writers &&
+    a.readwriters == b.readwriters &&
+    a.readUnderWrite == b.readUnderWrite
+  }  
```

Comment:
As you pointed out to me, you want to check everything but name. As I mentioned, you *could* (although perhaps shouldn't) create new memories with the same name and compare using the builtin ==, thus keeping this somewhat more robust to hopefully rare changes to the IR. I'm not sure
ie.
```scala
def eqMems(a: DefMemory, b: DefMemory) = a.copy(name = "blah") == b.copy(name = "blah")
```